rustyms-py/src/lib.rs

```diff
@@ -0,0 +1,1262 @@
+    ///
+    #[getter]
+    fn n_term(&self) -> Option<Modification> {
+        match &self.0.n_term {
+            Some(m) => Some(Modification(m.clone())),
+            None => None,
+        }
+    }
+
+    /// C-terminal modification.
+    ///
+    /// Returns
+    /// -------
+    /// Modification | None
+    ///
+    #[getter]
+    fn c_term(&self) -> Option<Modification> {
+        match &self.0.c_term {
+            Some(m) => Some(Modification(m.clone())),
+            None => None,
+        }
+    }
+
+    /// Sequence of the peptide including modifications.
+    ///
+    /// Returns
+    /// -------
+    /// list[SequenceElement]
+    ///
+    #[getter]
+    fn sequence(&self) -> Vec<SequenceElement> {
+        self.0
+            .sequence
+            .iter()
+            .map(|x| SequenceElement(x.clone()))
+            .collect()
+    }
+
+    /// For each ambiguous modification list all possible positions it can be placed on. Indexed by the ambiguous modification id.
+    ///
+    /// Returns
+    /// -------
+    /// list[list[int]]
+    ///
+    #[getter]
+    fn ambiguous_modifications(&self) -> Vec<Vec<usize>> {
+        self.0.ambiguous_modifications.clone()
+    }
+
+    /// Stripped sequence, meaning the sequence without any modifications.
+    ///
+    /// Returns
+    /// -------
+    /// str
+    ///
+    #[getter]
+    fn stripped_sequence(&self) -> String {
+        self.0.sequence.iter().map(|x| x.aminoacid.char()).collect()
+    }
+
+    // TODO: How to get the charge as an integer?
+    // /// The adduct ions, if specified.
+    // #[getter]
+    // fn charge_carriers(&self) -> Vec<> {
+    //     todo!()
+    // }
+
+    /// Get a copy of the peptide with its sequence reversed.
+    ///
+    /// Returns
+    /// -------
+    /// LinearPeptide
+    ///
+    fn reverse(&self) -> LinearPeptide {
+        LinearPeptide(self.0.reverse())
+    }
+
+    /// Gives the formula for the whole peptide. With the global isotope modifications applied.
+    ///
+    /// Returns
+    /// -------
+    /// MolecularFormula | None
+    ///
+    fn formula(&self) -> Option<MolecularFormula> {
+        match self.0.formula() {
+            Some(f) => Some(MolecularFormula(f)),
+            None => None,
+        }
+    }
+
+    /// Generate the theoretical fragments for this peptide, with the given maximal charge of the fragments, and the given model. With the global isotope modifications applied.
+    ///
+    /// Parameters
+    /// ----------
+    /// max_charge : int
+    ///    The maximal charge of the fragments.
+    /// model : str
+    ///   The model to use for the fragmentation. One of: ``all``, ``cid_hcd``, ``etcid``, ``etd``, ``ethcd``
+    ///
+    /// Returns
+    /// -------
+    /// list[Fragment]
+    ///   The theoretical fragments.
+    ///
+    fn generate_theoretical_fragments(
+        &self,
+        max_charge: i16,
+        model: &str,
+        // peptide_index: usize, TODO: Required for linear peptide?
```

Comment:
It is required for a linear one because the complex calls exactly the same function. Maybe it would be nicer indeed to write some in between method that auto fills 0 in the linear peptide case.